Sample_Notebooks/RFEP_analysis.ipynb

```diff
@@ -0,0 +1,175 @@
+    "\n",
+    "if 'traj_file' in global_conf:\n",
+    "    colvarsTraj = path + global_conf['traj_file']\n",
+    "else:\n",
+    "    colvarsTraj = path + global_conf['output_prefix'] + '.colvars.traj'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extract precomputed TI info from traj file\n",
+    "dAdL = TI_traj[rest_name]['dAdL']\n",
+    "lambdas =  TI_traj[rest_name]['L']\n",
+    "\n",
+    "# if lambdaExponent >=2, set a zero derivative for lambda=0 (might be NaN in the data)\n",
+    "if int(restraint_config['lambdaExponent']) >= 2 and np.isnan(dAdL[-1]):\n",
+    "    dAdL[-1] = 0.0\n",
+    "\n",
+    "if restraint_config['decoupling']: # lambdas have opposite meaning\n",
+    "    # Convert to coupling\n",
+    "    dAdL = - np.array(dAdL)\n",
+    "    lambdas = 1.0 - np.array(lambdas)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Setup and processing of colvars data\n",
+    "with open(colvarsTraj) as f:\n",
+    "    first_line = f.readline()\n",
+    "columns = re.split(' +', first_line)[1:-1]\n",
+    "\n",
+    "# trajectory could be read using colvartools (read multiple files etc)\n",
+    "# this might fail if vector variables are present\n",
+    "dataTI = pd.read_csv(colvarsTraj, delim_whitespace=True, names=columns, comment='#', index_col=0) \n",
+    "\n",
+    "# Process the harmonic wall restraint\n",
+    "DBC_rest = safep.make_harmonicWall_from_Colvars(restraint_config)\n",
+    "nLambdas = DBC_rest['targetNumStages'] + 1\n",
+    "\n",
+    "# We could also take a user parameter to adjust this in post-processing, or do equilibration detection\n",
+    "n_equil = DBC_rest['targetEquilSteps']\n",
+    "\n",
+    "cv =  DBC_rest['colvar']\n",
+    "dataTI = dataTI.rename(columns={cv:'DBC'})\n",
+    "\n",
+    "if DBC_rest['schedule'] == []:\n",
+    "    schedule = np.linspace(0, 1, nLambdas)\n",
+    "else:\n",
+    "    schedule = DBC_rest['schedule']\n",
+    "\n",
+    "dataTI = dataTI[dataTI.index>=n_equil][1:] # Remove first samples of each window from analysis\n",
+    "dataTI.index = dataTI.index-n_equil\n",
+    "\n",
+    "Ls = (dataTI.index.values-1)//DBC_rest['targetNumSteps']\n",
+    "\n",
+    "dataLs = np.round([schedule[i] for i in Ls], 3)\n",
+    "dataTI.loc[:,'L'] = dataLs\n",
+    "dataTI = dataTI.iloc[1:]\n",
+    "\n",
+    "TIperWindow, TIcumulative = safep.process_TI(dataTI, DBC_rest, None)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dG_DBC = np.round(TIcumulative['dG'][1], 1)\n",
+    "error_DBC = np.round(TIcumulative['error'][1], 1)\n",
+    "change_mkd_TI = f'\\u0394G<sub>DBC</sub> = {dG_DBC} kcal/mol'\n",
+    "error_mkd_TI = f'Standard Deviation: {error_DBC} kcal/mol'\n",
+    "print(change_mkd_TI, error_mkd_TI)"
```

Comment:
To print markdown in a jupyter notebook:
```
from IPython.display import display, Markdown
...
display(Markdown(change_mkd_TI))
display(Markdown(error_mkd_TI))
```
Otherwise, this gives me:
`ΔG<sub>DBC</sub> = -0.9 kcal/mol Standard Deviation: 0.4 kcal/mol`

Reply:
Thanks, that was carried over from the tutorial, I've simplified it by removing the markdown.